pkg/dependency/parser/python/uv/parse.go

```diff
@@ -0,0 +1,166 @@
+	if _, ok := visited[pkg.Name]; ok {
+		return
+	}
+	visited[pkg.Name] = struct{}{}
+	for _, dep := range pkg.Dependencies {
+		depPkg, exists := packages[dep.Name]
+		if !exists {
+			continue
+		}
+		walkPackageDeps(depPkg, packages, visited)
+	}
+}
+
+func (l Lock) root() Package {
+	for _, pkg := range l.Packages {
+		if pkg.isRoot() {
+			return pkg
+		}
+	}
+
+	return Package{}
+}
+
+type Package struct {
+	Name         string       `toml:"name"`
+	Version      string       `toml:"version"`
+	Source       Source       `toml:"source"`
+	Dependencies []Dependency `toml:"dependencies"`
+}
+
+// https://github.com/astral-sh/uv/blob/f7d647e81d7e1e3be189324b06024ed2057168e6/crates/uv-resolver/src/lock/mod.rs#L572-L579
+func (p Package) isRoot() bool {
+	return p.Source.Editable == "." || p.Source.Virtual == "."
+}
+
+type Source struct {
+	Editable string `toml:"editable"`
+	Virtual  string `toml:"virtual"`
+}
+
+type Dependency struct {
+	Name string `toml:"name"`
+}
+
+type Parser struct{}
+
+func NewParser() *Parser {
+	return &Parser{}
+}
+
+func (p *Parser) Parse(r xio.ReadSeekerAt) ([]ftypes.Package, []ftypes.Dependency, error) {
+	var lock Lock
+	if _, err := toml.NewDecoder(r).Decode(&lock); err != nil {
+		return nil, nil, xerrors.Errorf("failed to decode uv lock file: %w", err)
+	}
+
+	rootPackage := lock.root()
+	if rootPackage.Name == "" {
+		return nil, nil, xerrors.New("uv lockfile does not contain a root package. " +
+			"See https://github.com/astral-sh/uv/blob/f80ddf10b63c3e7b421ca4658e63f97db1e0378c/crates/uv/src/commands/project/lock.rs#L933-L936")
+	}
+
+	packages := lock.packages()
+	directDeps := lock.directDeps()
+	prodDeps := lock.prodDeps()
+
+	var (
+		pkgs []ftypes.Package
+		deps []ftypes.Dependency
+	)
+
+	for _, pkg := range lock.Packages {
+		if _, ok := prodDeps[pkg.Name]; !ok {
+			continue
+		}
+
+		pkgID := packageID(pkg.Name, pkg.Version)
+		relationship := ftypes.RelationshipIndirect
+		if pkg.isRoot() {
+			relationship = ftypes.RelationshipRoot
+		} else if _, ok := directDeps[pkg.Name]; ok {
+			relationship = ftypes.RelationshipDirect
+		}
+
+		pkgs = append(pkgs, ftypes.Package{
+			ID:           pkgID,
+			Name:         pkg.Name,
+			Version:      pkg.Version,
+			Indirect:     relationship == ftypes.RelationshipIndirect,
+			Relationship: relationship,
+		})
+
+		dependsOn := make([]string, 0, len(pkg.Dependencies))
+
+		for _, dep := range pkg.Dependencies {
+			depPkg, exists := packages[dep.Name]
+			if !exists {
+				continue
+			}
+			dependsOn = append(dependsOn, packageID(dep.Name, depPkg.Version))
+		}
+
+		if len(dependsOn) > 0 {
+			sort.Strings(dependsOn)
+			deps = append(deps, ftypes.Dependency{
+				ID:        pkgID,
+				DependsOn: dependsOn,
+			})
+		}
+	}
+
+	sort.Slice(pkgs, func(i, j int) bool {
+		return pkgs[i].ID < pkgs[j].ID
+	})
+
+	sort.Slice(deps, func(i, j int) bool {
+		return deps[i].ID < deps[j].ID
+	})
```

Comment:
```suggestion
	sort.Sort(pkgs)
	sort.Sort(deps)
```

But you need to use `Packages` and `Dependencies` here:
https://github.com/aquasecurity/trivy/blob/8ee3c36585497abe40920c01f3b38b8618e1668e/pkg/dependency/parser/python/uv/parse.go#L109-L110

Reply:
I forgot to look at the sorting implementations :)